src/Lykke.Job.TransactionHandler/Projections/OperationHistoryProjection.cs

```diff
@@ -55,8 +58,71 @@ public class OperationHistoryProjection
             _walletCredentialsRepository = walletCredentialsRepository ?? throw new ArgumentNullException(nameof(walletCredentialsRepository));
             _clientCacheRepository = clientCacheRepository ?? throw new ArgumentNullException(nameof(clientCacheRepository));
             _feeLogService = feeLogService ?? throw new ArgumentNullException(nameof(feeLogService));
+            _ethereumTransactionRequestRepository = ethereumTransactionRequestRepository ?? throw new ArgumentNullException(nameof(ethereumTransactionRequestRepository));
         }
-        
+
+
+        public async Task Handle(TransferOperationStateSavedEvent evt)
+        {
+            await _log.WriteInfoAsync(nameof(OperationHistoryProjection), nameof(TransferOperationStateSavedEvent), evt.ToJson(), "");
+
+            var message = evt.QueueMessage;
+            var transactionId = message.Id;
+            var amountNoFee = evt.AmountNoFee;
+
+            var context = await _transactionService.GetTransactionContext<TransferContextData>(transactionId);
+
+            //Get eth request if it is ETH transfer
+            var ethTxRequest = await _ethereumTransactionRequestRepository.GetAsync(Guid.Parse(transactionId));
+
+            //Get client wallets
+            var destWallet = await _walletCredentialsRepository.GetAsync(message.ToClientid);
+            var sourceWallet = await _walletCredentialsRepository.GetAsync(message.FromClientId);
```

Comment:
Will be good to add request with clientId collections in the repository for this case, to request client wallets with single round trip to the table storage